crates/wash-lib/src/app.rs

```diff
@@ -50,10 +52,59 @@ impl ToString for ModelOperation {
 
 #[derive(Debug)]
 pub enum AppManifest {
-    SerializedModel(String),
+    SerializedModel(serde_yaml::Value),
     ModelName(String),
 }
 
+impl AppManifest {
+    /// Resolve relative file paths in the given app manifest to some base path
+    pub fn resolve_image_relative_file_paths(&mut self, base: impl AsRef<Path>) -> Result<()> {
+        if let AppManifest::SerializedModel(ref mut content) = self {
+            resolve_relative_file_paths_in_yaml(content, base)?;
+        }
+        Ok(())
+    }
+}
+
+/// Resolve the relative paths in a YAML value, given a base path (directory)
+/// from which to resolve the relative paths that are found
+fn resolve_relative_file_paths_in_yaml(
+    content: &mut serde_yaml::Value,
+    base_dir: impl AsRef<Path>,
+) -> Result<()> {
+    match content {
+        // If we encounter a string anywhere that is a relative path, resolve it
+        serde_yaml::Value::String(s)
+            if s.starts_with("file://") && s.chars().nth(7).is_some_and(|v| v != '/') =>
+        {
+            // Convert the base dir + relative path into a file based URL
+            let full_path = base_dir.as_ref().join(
+                s.strip_prefix("file://")
+                    .context("failed to strip prefix")?,
+            );
+            // Build a file based URL and replace the existing one
+            if let Ok(url) = Url::from_file_path(full_path) {
+                *s = url.into();
+            }
+        }
```

Comment:
Thoughts on outputting a warning log or similar for when we resolve a full file path, and then nothing is there?

This may or may not be an error depending on where the dev is in their workflow of course, but just a little heads up to say "hey, I know you wanted this Wasm and I didn't find it"

Reply:
Sure, added that! It's possible that it's not a WASM file, could be a par.gz or any other kind of file actually but I added a more vague warning

Reply:
It ideally *shouldn't* happen, but previous code does not outright reject file paths that don't exist, so I didn't want to make that change here.

"reject file paths that don't exist" is definitely a worthwhile change, but can be handled in a follow up -- it's definitely a change of functionality.